scikeras/wrappers.py

```diff
@@ -706,9 +710,10 @@ def fit(self, X, y, sample_weight=None, **kwargs) -> "BaseWrapper":
             A reference to the instance that can be chain called
             (ex: instance.fit(X,y).transform(X) )
         """
-        for k, v in kwargs.items():
-            warnings.warn(_kwarg_warn.format("fit", k, v))
-            break  # only warn about the first kwarg
+        if kwargs:
+            example_kwarg = next(iter(kwargs.keys()))
+            kwarg_list = "`\n * `".join(kwargs.keys()) + "`"
```

Comment:
Shouldn't the error message display the value too?

Reply:
What are the values needed for? I don't see how a developer would use that to fix the issue, they can just look up the value in their code.

Reply:
Why not show the values? There's no real downside, especially because you're showing each value on a newline at most initialization parameters are basic types (int, str, float, etc).

Reply:
> There's no real downside

I think the main downside for long/large warnings is that the information gets lost in the noise. Repeated 20+ line warnings can make it hard to know what is even important.

> most initialization parameters are basic types

You are right, most are basic types. But several, like loss, optimizer, metrics, callbacks, etc. are at the very least iterables/dicts and at the very worst dicts of instances.

Reply:
> Repeated 20+ line warnings can make it hard to know what is even important.

Repetition is not an issue. From the [warnings docs][1], "Repetitions of a particular warning for the same source location are typically suppressed." Plus, this warning is only issued once per `fit`/`predict` call (which aren't repeated very often; I would say `partial_fit` is repeated frequently).

I think having one 20+ line warning is okay. I think this warning will be short (<= 4 lines) the vast majority of times it's issued.


[1]:https://docs.python.org/3/library/warnings.html

Reply:
You make some good points. Besides, we are currently printing these messages with the values, etc, so it's certainly not going to be _worse_ than what we currently have.

I've pushed a commit to print the values, as well as give examples using fixed dummy `foo=bar` kwargs (https://github.com/adriangb/scikeras/pull/198#discussion_r577793274).